src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -56,12 +58,37 @@ def _get_missing_members(self, organization: Organization) -> QuerySet[CommitAut
             .order_by("-commit_count")
         )
 
-    # TODO(cathy): check domain
+    def _get_domain_from_email(self, email: str) -> str:
+        return email.split("@")[1]
 
     def get(self, request: Request, organization) -> Response:
-        # TODO(cathy): search
         queryset = self._get_missing_members(organization)
 
+        # if a member has user_email=None, then they have yet to accept an invite
+        org_owners = organization.get_members_with_org_roles(
+            roles=[roles.get_top_dog().id]
+        ).exclude(user_email=None)
+
+        prev_domain = self._get_domain_from_email(org_owners[0].user_email)
+        common_domain = True
+        for org_owner in org_owners:
+            if prev_domain and self._get_domain_from_email(org_owner.user_email) != prev_domain:
+                common_domain = False
+                break
+
+        if common_domain and prev_domain:
+            queryset = queryset.filter(email__endswith=prev_domain)
+
+        query = request.GET.get("query")
+        if query:
+            tokens = tokenize_query(query)
+            for key, value in tokens.items():
+                if key == "query":
+                    query_value = " ".join(value)
+                    queryset = queryset.filter(
+                        Q(email__icontains=query_value) | Q(external_id__icontains=query_value)
+                    )
```

Comment:
`tokens` should only have one "query" key
```suggestion
            if "query" in tokens:
                query_value = " ".join(tokens["query"])
                queryset = queryset.filter(
                    Q(email__icontains=query_value) | Q(external_id__icontains=query_value)
                )
```

Reply:
it's possible that there may be different kinds of tokens in the future (like to filter for a single integration vs getting them all). should i leave it as a loop or refactor it when the time comes?

Reply:
Do you mean in the future you might be filter using additional keys other than `query`?

Reply:
yes

Reply:
In that case would it not be cleaner to check for individual keys you're filtering on instead of looping over the whole dictionary